base/strings/search.jl

```diff
@@ -442,6 +442,8 @@ function _rsearchindex(s::String, t::String, i::Integer)
     # Check for fast case of a single byte
     if lastindex(t) == 1
         return something(findprev(isequal(t[1]), s, i), 0)
+    elseif sizeof(t) > sizeof(s)
+        return 0
     elseif lastindex(t) != 0
         j = i ≤ ncodeunits(s) ? nextind(s, i)-1 : i
         return _rsearchindex(unsafe_wrap(Vector{UInt8}, s), unsafe_wrap(Vector{UInt8}, t), j)
```

Comment:
It seems like `_rsearchindex` should be fixed for byte arrays, not just for `String`.  i.e. you are fixing the wrong method.

Reply:
It seems to me the byte array code should error and we can't fix it there without substantial re-factoring.

The thing looks to me people expect `find(string...)` to return `nothing` when the same situation should `error`:

```
julia> findlast("xxx", "")

julia> findlast(UInt8[1,2,3], UInt8[])
ERROR: BoundsError: attempt to access 0-element Vector{UInt8} at index [0]
```

I personally think the first one should error too, but 1.5 didn't have that behavior thus we're fixing this bug

Reply:
https://github.com/JuliaLang/julia/blob/bc3ce48636b700d153d2c84614aaea8cb8843d43/base/strings/search.jl#L462

simply changing this to `k<0` (without any other changes in this PR) seems to pass all the test (old + new), what you think? @stevengj